rpcs3/Emu/Audio/Cubeb/CubebBackend.cpp

```diff
@@ -413,36 +486,52 @@ void CubebBackend::state_cb(cubeb_stream* /* stream */, void* user_ptr, cubeb_st
 	}
 }
 
-void CubebBackend::device_collection_changed_cb(cubeb* /* context */, void* user_ptr)
+void CubebBackend::device_collection_changed_cb(cubeb* context, void* user_ptr)
 {
+	Cubeb.notice("Device collection changed");
+
+	if (!context)
+	{
+		Cubeb.error("device_collection_changed_cb called with invalid stream");
+		return;
+	}
+
 	CubebBackend* const cubeb = static_cast<CubebBackend*>(user_ptr);
+	ensure(cubeb);
+
+	if (context != cubeb->m_ctx)
```

Comment:
Must be done under the lock

Reply:
m_ctx does not seem to be locked anywhere in the class

Reply:
I guess locking ctor with m_dev_sw_mutex would suffice